### PR TITLE
path: Add Components iterator

### PR DESCRIPTION
**Pull request by nicholasbishop**

### CODE REVIEW

File: src/path.rs [hunk 1]
```diff
@@ -388,6 +403,62 @@ where
...
+        {
+            self.offset + index
+        } else {
+            path.len()
+        };
+
+        let component = &path[self.offset..end];
+        let component = if component == b".." {
+            Component::ParentDir
+        } else {
```
Comment (tedbrandston) [inline src/path.rs:450]: What about CurDir?

File: src/path.rs [hunk 2]
```diff
@@ -595,4 +666,64 @@ mod tests {
...
+        let p = PathBuf::new("/a/b");
+        let c: Vec<_> = p.components().collect();
+        assert_eq!(
+            c,
+            [
+                Component::RootDir,
+                Component::normal("a").unwrap(),
+                Component::normal("b").unwrap(),
+            ]
+        );
```
Comment (tedbrandston) [inline src/path.rs:727]: Similarly: tests for CurDir? Even if we want `.` to be returned as a `Normal` I feel like there should be tests to express that.

File: src/path.rs (hunk 1 above)
Comment (nicholasbishop) [reply src/path.rs:452]: Good catch. I added `CurDir` after I wrote the initial `Components` implementation and forgot to update it. Added it.

File: src/path.rs (hunk 2 above)
Comment (nicholasbishop) [reply src/path.rs:730]: Added

**Review by tedbrandston — APPROVED** [review]

**Pull request merged**